ExploreASL_Initialize.m

```diff
@@ -27,287 +21,249 @@
-    SelectParFile = true;
-elseif exist(DataParPath,'dir')
-    SelectParFile = true;
-    warning('Data parameter file was inputted as folder, but should point to a file instead');
-else
-    SelectParFile = false;
-end
-
-x.S = struct;
 
-if nargin>1 && ~isempty(ProcessData)
-    if ProcessData==0 || ProcessData==1  || ProcessData==2
-        x.ProcessData = ProcessData; % skip GUI
+    % Check if the DataParPath is a file or a directory
+    SelectParFile = false; % Fallback
+    if x.ProcessData
+        % Checkout the "Proceed with Initialization" section
+        if (isempty(x.DataParPath) || ~exist(x.DataParPath,'file'))
+            SelectParFile = true; % If the DataParPath is either empty OR the file does not exist, we have to select it later on (if processing is turned on)
+        end
+    end
+    
+    % Check if GUI options are available
+    if ~usejava('desktop') || ~usejava('jvm') || ~feature('ShowFigureWindows')
+        bUseGUI = false;
     else
-        warning('Invalid input argument "ProcessData", should be 0, 1, or 2, resetting');
+        bUseGUI = true;
     end
-end
-
+    
 
-%% Question dialogue - initialize only or run data
- if ~usejava('desktop') || ~usejava('jvm') || ~feature('ShowFigureWindows')
-     bUseGUI = false;
- else
-     bUseGUI = true;
- end
+    %% -----------------------------------------------------------------------------
+    %% Get ExploreASL path
 
-if ~isfield(x,'ProcessData')
+    % Check if the current directory is the ExploreASL directory
+    CurrCD = pwd;
+    if exist(fullfile(CurrCD, 'ExploreASL_Master.m'), 'file')
+        x.MyPath = CurrCD;
+    end
 
-    InitChoice = 'Process dataset';
-    if ~exist('DataParPath','var')
-        ParFileExists = false;
-    elseif isempty(DataParPath)
-        ParFileExists = false;
-    else
-        ParFileExists = true;
+    % Check whether MyPath is correct, otherwise obtain correct folder
+    if ~isfield(x, 'MyPath')
+        x.MyPath = '/DummyPath';
     end
 
-    if ~ParFileExists
-        if bUseGUI
-            InitChoice = questdlg('Would you like to initialize ExploreASL functionality, load a dataset and/or start the processing pipeline?', ...
-                'Start up ExploreASL', 'Process dataset', 'Only initialize ExploreASL functionality', 'Load dataset only', 'Only initialize ExploreASL functionality');
-            if isempty(InitChoice)
-                InitChoice = 'Cancel';
+    % Get the master script path
+    MasterScriptPath = fullfile(x.MyPath, 'ExploreASL_Master.m');
+
+    % Select the ExploreASL folder manually, if the script is not run in deployed mode
+    if ~isdeployed
+        if ~exist(MasterScriptPath,'file')
+            if bUseGUI
+                if ismac % somehow macOS doesnt show the title of the question dialogue
+                    fprintf('Select folder where ExploreASL is installed\n');
+                end
+                pathstr = uigetdir(CurrCD, 'Select folder where ExploreASL is installed');
```

Comment:
We decided to get rid of the GUI stuff here right, then I would also always use the `input` here, irregardless of OS.

Reply:
Sure, makes my life easier :)
I thought you only wanted to get rid of the questdlg boxes and not of the uigetdir ones.

Reply:
Removed it: e68fa0d.